### PR TITLE
feat(1271): add compact query to templates and commands

### PR DESCRIPTION
**Pull request by DekusDenial**
## Context

API call to `/templates` is way too slow for the node.js layer to serialize all the bloated json payload

## Objective

- Apply the same compact query optimization to `templates` and `commands` to speed up the API performance by over 95%
- Trimmed the fat off the response payload
- Dedupe multi versions by grouping on `namespace` and `name`

## References

related to: screwdriver-cd/screwdriver#1271

### CODE REVIEW

File: plugins/templates/list.js [hunk 1]
```diff
@@ -58,18 +67,23 @@ module.exports = () => ({
...
                 };
             }
 
-            if (request.query.page || request.query.count) {
-                config.paginate = {
-                    page: request.query.page,
-                    count: request.query.count
-                };
+            // check if the call wants compact data
+            if (compact === '1') {
```
Comment (tkyi) [inline plugins/templates/list.js]: why `1` vs `true`/`false`?

**Comment by tkyi** [comment]
Some banner test failures probably due to https://github.com/screwdriver-cd/data-schema/pull/295/files

File: plugins/templates/list.js (hunk 1 above)
Comment (tkyi) [reply plugins/templates/list.js]: I don't have a strong preference either way
I think we usually stick to `true`/`false`

Comment (DekusDenial) [reply plugins/templates/list.js]: i can change

Comment (DekusDenial) [reply plugins/templates/list.js]: kk, will change

**Comment by DekusDenial** [comment]
@tkyi i can remove those tests since they are no longer relevant (due to the change that removed the defaults from the schema)

**Pull request merged**